### PR TITLE
Enable and fix gosec lint

### PR DESCRIPTION
**Pull request by futuretea**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

linter: [gosec](https://github.com/securego/gosec) - Inspects source code for security problems by scanning the Go AST.
rule: [G601 - Implicit memory aliasing of items from a range statement](https://github.com/securego/gosec#usage)

**Related Issue:**
https://github.com/harvester/harvester/issues/4019

**Test plan:**

### CODE REVIEW

**Comment by futuretea** [comment]
It should be noted that gosec rule G601 currently has some cases that cannot be recognized.
related issue: https://github.com/securego/gosec/issues/966
related PR: https://github.com/securego/gosec/pull/967

File: .golangci.json [hunk 1]
```diff
@@ -26,6 +27,11 @@
 	"linters-settings": {
 		"goimports": {
 			"local-prefixes": "github.com/harvester/harvester"
+		},
+		"gosec": {
+			"includes": [
+				"G601"
```
Comment (guangbochen) [inline .golangci.json]: is there any specific reason why we only need the G601 or we should include all rules by default?

Comment (yaocw2020) [inline .golangci.json]: Is it mandatory to add？

Comment (futuretea) [reply .golangci.json]: only enable G601 rule

Comment (futuretea) [reply .golangci.json]: There is the output if we enable all gosec rules:
```bash
❯ golangci-lint run
tests/framework/fuzz/port.go:19:15: G102: Binds to all network interfaces (gosec)
                lis, err := net.Listen("tcp", ":0")
                            ^
pkg/api/proxy/handler.go:47:33: G402: TLS InsecureSkipVerify set true. (gosec)
                        TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
                                                     ^
tests/framework/helper/http.go:19:34: G402: TLS InsecureSkipVerify set true. (gosec)
                                TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
                                                             ^
pkg/server/ui/ui.go:24:5: G402: TLS InsecureSkipVerify set true. (gosec)
                                InsecureSkipVerify: true,
                                ^
pkg/codegen/main.go:163:11: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        if err = ioutil.WriteFile(absPath, output, 0644); err != nil {
                 ^
pkg/codegen/main.go:179:11: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        if err = ioutil.WriteFile(absPath, output, 0644); err != nil {
                 ^
pkg/codegen/main.go:195:11: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        if err = ioutil.WriteFile(absPath, output, 0644); err != nil {
                 ^
pkg/util/constants.go:34:2: G101: Potential hardcoded credentials (gosec)
        InternalTLSSecretName             = "tls-rancher-internal"
        ^
pkg/cmd/app.go:83:16: G114: Use of net/http serve function that has no support for setting timeouts (gosec)
                        log.Println(http.ListenAndServe(options.ProfilerAddress, nil))
                                    ^
pkg/controller/master/storagenetwork/storage_network.go:5:2: G505: Blocklisted import crypto/sha1: weak cryptographic primitive (gosec)
        "crypto/sha1"
        ^
pkg/controller/master/storagenetwork/storage_network.go:256:10: G401: Use of weak cryptographic primitive (gosec)
        hash := sha1.New()
                ^
pkg/webhook/resources/upgrade/validator.go:30:2: G101: Potential hardcoded credentials (gosec)
        skipWebhookAnnotation = "harvesterhci.io/skipWebhook"
        ^
pkg/controller/master/backup/backup_target.go:34:2: G101: Potential hardcoded credentials (gosec)
        AWSSecretKey       = "AWS_SECRET_ACCESS_KEY"
        ^
pkg/webhook/resources/setting/validator.go:399:38: G402: TLS MinVersion too low. (gosec)
                customTransport.TLSClientConfig = &tls.Config{
                        RootCAs: certs,
                }
main.go:11:2: G108: Profiling endpoint is automatically exposed on /debug/pprof (gosec)
        _ "net/http/pprof"
        ^
cmd/webhook/main.go:5:2: G108: Profiling endpoint is automatically exposed on /debug/pprof (gosec)
        _ "net/http/pprof"
        ^
```

Comment (futuretea) [reply .golangci.json]: It seems that the warnning of the other rules are tolerable

Comment (guangbochen) [reply .golangci.json]: can we help to track those issue and add the `exclude` list first, WDUT?

**Review by yaocw2020 — APPROVED** [review]
LGTM, thanks

File: .golangci.json (hunk 1 above)
Comment (futuretea) [reply .golangci.json]: I will update this PR

Comment (futuretea) [reply .golangci.json]: updated

**Review by guangbochen — APPROVED** [review]
LGTM, and thanks for the fix and new golangci

**Review by bk201 — APPROVED** [review]
Thanks, good enhancement!

**Comment by guangbochen** [comment]
@mergify backport v1.2

**Pull request merged**